### ISSUE TITLE
ext clock broken

### ISSUE THREAD

**Issue by tehn** [tehn/awake#1]

**Comment by okyeron** [comment]
For the `on_select_external ` error the following change seems to fix the error. Is just the pattern reset the intended goal there?

[line 156](https://github.com/tehn/awake/blob/0d5e4ba0204579ea4fa154671bed12e834d11945/awake.lua#L156):
```
clk.on_select_external = function() clk:reset() end --reset_pattern
```

Otherwise I'm looking at beatclock.lua with regards to external clock as I'm not certain it's working properly there.

**Comment by okyeron** [comment]
Also a dumb question: I am adding a bunch of debug print statements in beatclock, but none of them show up in the REPL at all. Is there something funny here because it's a "required" module? 

Any trick to get those prints in beatclock to work?

**Comment by okyeron** [comment]
Ok - I think I sorted out the external clock problem. (turns out my prints were not happening because `process_midi` wasn't actually executing)

There are 2 midi.connect() statements in awake. [Line 80](https://github.com/tehn/awake/blob/0d5e4ba0204579ea4fa154671bed12e834d11945/awake.lua#L80) and [Line 150](https://github.com/tehn/awake/blob/0d5e4ba0204579ea4fa154671bed12e834d11945/awake.lua#L150)

As they are, they are both trying to connect to vport 1 - I think the second one is over-riding the first one. At which point `process_midi` is just not happening. Maybe because the one in init() is blocking the one for `process_midi`?

If I set `midi_out_device` to `midi.connect(2)` then process_midi (and external clock) works again.

So... not sure if this is a larger problem - maybe needing some debug if a script tries to connect to the same midi port twice ?
